################################################################################
Concuerror started with options:
  [{after_timeout,infinity},
   {assume_racing,false},
   {delay_bound,infinity},
   {depth_bound,5000},
   {entry_point,{after_test,after_test,[]}},
   {files,["/home/stavros/git/Concuerror/tests/suites/dpor_tests/src/after_test.erl"]},
   {ignore_error,[]},
   {ignore_first_crash,true},
   {instant_delivery,false},
   {non_racing_system,[]},
   {optimal,true},
   {print_depth,20},
   {quiet,true},
   {scheduling,round_robin},
   {show_races,false},
   {strict_scheduling,false},
   {symbolic_names,true},
   {timeout,infinity},
   {treat_as_normal,[]}]
################################################################################
Erroneous interleaving 1:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   5: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   6: P.1: exits normally
   7: Message (ok1) from P.1 reaches P
   8: P: receives message (ok1)
    in after_test.erl line 22
   9: P: receive timeout expired after 0 ms
    in after_test.erl line 33
  10: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  11: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  12: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  13: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  14: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  15: P.2: exits normally
  16: Message (ok2) from P.2 reaches P
  17: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  18: P.3: exits normally
  19: Message (ok2) from P.3 reaches P
  20: P: receives message (ok2)
    in after_test.erl line 22
  21: P: receives message (ok2)
    in after_test.erl line 22
  22: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 2:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   5: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   6: P.1: exits normally
   7: Message (ok1) from P.1 reaches P
   8: P: receives message (ok1)
    in after_test.erl line 22
   9: P: receive timeout expired after 0 ms
    in after_test.erl line 33
  10: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  11: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  12: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  13: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  14: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  15: P.2: exits normally
  16: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  17: P.3: exits normally
  18: Message (ok2) from P.3 reaches P
  19: P: receives message (ok2)
    in after_test.erl line 22
  20: Message (ok2) from P.2 reaches P
  21: P: receives message (ok2)
    in after_test.erl line 22
  22: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 3:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   5: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   6: P.1: exits normally
   7: Message (ok1) from P.1 reaches P
   8: P: receives message (ok1)
    in after_test.erl line 22
   9: P: receive timeout expired after 0 ms
    in after_test.erl line 33
  10: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  11: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  12: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  13: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  14: P.2: exits normally
  15: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  16: P.3: exits normally
  17: Message (ok2) from P.2 reaches P
  18: Message (ok2) from P.3 reaches P
  19: P: receives message (ok2)
    in after_test.erl line 26
  20: P: receives message (ok2)
    in after_test.erl line 26
  21: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 4:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   5: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   6: P.1: exits normally
   7: Message (ok1) from P.1 reaches P
   8: P: receives message (ok1)
    in after_test.erl line 22
   9: P: receive timeout expired after 0 ms
    in after_test.erl line 33
  10: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  11: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  12: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  13: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  14: P.2: exits normally
  15: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  16: P.3: exits normally
  17: Message (ok2) from P.2 reaches P
  18: P: receives message (ok2)
    in after_test.erl line 26
  19: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  20: Message (ok2) from P.3 reaches P
  21: P: receives message (ok2)
    in after_test.erl line 22
  22: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 5:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   5: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   6: P.1: exits normally
   7: Message (ok1) from P.1 reaches P
   8: P: receives message (ok1)
    in after_test.erl line 22
   9: P: receive timeout expired after 0 ms
    in after_test.erl line 33
  10: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  11: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  12: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  13: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  14: P.2: exits normally
  15: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  16: P.3: exits normally
  17: Message (ok2) from P.2 reaches P
  18: P: receives message (ok2)
    in after_test.erl line 26
  19: Message (ok2) from P.3 reaches P
  20: P: receives message (ok2)
    in after_test.erl line 26
  21: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 6:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   5: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   6: P.1: exits normally
   7: Message (ok1) from P.1 reaches P
   8: P: receives message (ok1)
    in after_test.erl line 22
   9: P: receive timeout expired after 0 ms
    in after_test.erl line 33
  10: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  11: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  12: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  13: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  14: P.2: exits normally
  15: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  16: P.3: exits normally
  17: Message (ok2) from P.3 reaches P
  18: P: receives message (ok2)
    in after_test.erl line 26
  19: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  20: Message (ok2) from P.2 reaches P
  21: P: receives message (ok2)
    in after_test.erl line 22
  22: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 7:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   5: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   6: P.1: exits normally
   7: Message (ok1) from P.1 reaches P
   8: P: receives message (ok1)
    in after_test.erl line 22
   9: P: receive timeout expired after 0 ms
    in after_test.erl line 33
  10: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  11: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  12: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  13: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  14: P.2: exits normally
  15: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  16: P.3: exits normally
  17: Message (ok2) from P.3 reaches P
  18: P: receives message (ok2)
    in after_test.erl line 26
  19: Message (ok2) from P.2 reaches P
  20: P: receives message (ok2)
    in after_test.erl line 26
  21: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 8:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   5: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   6: P.1: exits normally
   7: Message (ok1) from P.1 reaches P
   8: P: receives message (ok1)
    in after_test.erl line 22
   9: P: receive timeout expired after 0 ms
    in after_test.erl line 33
  10: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  11: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  12: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  13: P.2: exits normally
  14: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  15: P.3: exits normally
  16: Message (ok2) from P.2 reaches P
  17: Message (ok2) from P.3 reaches P
  18: P: receives message (ok2)
    in after_test.erl line 26
  19: P: receives message (ok2)
    in after_test.erl line 26
  20: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 9:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   5: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   6: P.1: exits normally
   7: Message (ok1) from P.1 reaches P
   8: P: receives message (ok1)
    in after_test.erl line 22
   9: P: receive timeout expired after 0 ms
    in after_test.erl line 33
  10: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  11: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  12: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  13: P.2: exits normally
  14: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  15: P.3: exits normally
  16: Message (ok2) from P.2 reaches P
  17: P: receives message (ok2)
    in after_test.erl line 26
  18: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  19: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  20: Message (ok2) from P.3 reaches P
  21: P: receives message (ok2)
    in after_test.erl line 22
  22: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 10:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   5: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   6: P.1: exits normally
   7: Message (ok1) from P.1 reaches P
   8: P: receives message (ok1)
    in after_test.erl line 22
   9: P: receive timeout expired after 0 ms
    in after_test.erl line 33
  10: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  11: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  12: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  13: P.2: exits normally
  14: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  15: P.3: exits normally
  16: Message (ok2) from P.2 reaches P
  17: P: receives message (ok2)
    in after_test.erl line 26
  18: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  19: Message (ok2) from P.3 reaches P
  20: P: receives message (ok2)
    in after_test.erl line 26
  21: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 11:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   5: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   6: P.1: exits normally
   7: Message (ok1) from P.1 reaches P
   8: P: receives message (ok1)
    in after_test.erl line 22
   9: P: receive timeout expired after 0 ms
    in after_test.erl line 33
  10: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  11: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  12: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  13: P.2: exits normally
  14: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  15: P.3: exits normally
  16: Message (ok2) from P.2 reaches P
  17: P: receives message (ok2)
    in after_test.erl line 26
  18: Message (ok2) from P.3 reaches P
  19: P: receives message (ok2)
    in after_test.erl line 26
  20: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 12:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   5: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   6: P.1: exits normally
   7: Message (ok1) from P.1 reaches P
   8: P: receives message (ok1)
    in after_test.erl line 22
   9: P: receive timeout expired after 0 ms
    in after_test.erl line 33
  10: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  11: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  12: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  13: P.2: exits normally
  14: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  15: P.3: exits normally
  16: Message (ok2) from P.3 reaches P
  17: P: receives message (ok2)
    in after_test.erl line 26
  18: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  19: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  20: Message (ok2) from P.2 reaches P
  21: P: receives message (ok2)
    in after_test.erl line 22
  22: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 13:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   5: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   6: P.1: exits normally
   7: Message (ok1) from P.1 reaches P
   8: P: receives message (ok1)
    in after_test.erl line 22
   9: P: receive timeout expired after 0 ms
    in after_test.erl line 33
  10: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  11: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  12: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  13: P.2: exits normally
  14: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  15: P.3: exits normally
  16: Message (ok2) from P.3 reaches P
  17: P: receives message (ok2)
    in after_test.erl line 26
  18: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  19: Message (ok2) from P.2 reaches P
  20: P: receives message (ok2)
    in after_test.erl line 26
  21: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 14:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   5: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   6: P.1: exits normally
   7: Message (ok1) from P.1 reaches P
   8: P: receives message (ok1)
    in after_test.erl line 22
   9: P: receive timeout expired after 0 ms
    in after_test.erl line 33
  10: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  11: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  12: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  13: P.2: exits normally
  14: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  15: P.3: exits normally
  16: Message (ok2) from P.3 reaches P
  17: P: receives message (ok2)
    in after_test.erl line 26
  18: Message (ok2) from P.2 reaches P
  19: P: receives message (ok2)
    in after_test.erl line 26
  20: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 15:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   5: P.1: exits normally
   6: Message (ok1) from P.1 reaches P
   7: P: receives message (ok1)
    in after_test.erl line 26
   8: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   9: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  10: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  11: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  12: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  13: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  14: P.2: exits normally
  15: Message (ok2) from P.2 reaches P
  16: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  17: P.3: exits normally
  18: Message (ok2) from P.3 reaches P
  19: P: receives message (ok2)
    in after_test.erl line 22
  20: P: receives message (ok2)
    in after_test.erl line 22
  21: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 16:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   5: P.1: exits normally
   6: Message (ok1) from P.1 reaches P
   7: P: receives message (ok1)
    in after_test.erl line 26
   8: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   9: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  10: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  11: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  12: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  13: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  14: P.2: exits normally
  15: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  16: P.3: exits normally
  17: Message (ok2) from P.3 reaches P
  18: P: receives message (ok2)
    in after_test.erl line 22
  19: Message (ok2) from P.2 reaches P
  20: P: receives message (ok2)
    in after_test.erl line 22
  21: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 17:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   5: P.1: exits normally
   6: Message (ok1) from P.1 reaches P
   7: P: receives message (ok1)
    in after_test.erl line 26
   8: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   9: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  10: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  11: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  12: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  13: P.2: exits normally
  14: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  15: P.3: exits normally
  16: Message (ok2) from P.2 reaches P
  17: Message (ok2) from P.3 reaches P
  18: P: receives message (ok2)
    in after_test.erl line 26
  19: P: receives message (ok2)
    in after_test.erl line 26
  20: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 18:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   5: P.1: exits normally
   6: Message (ok1) from P.1 reaches P
   7: P: receives message (ok1)
    in after_test.erl line 26
   8: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   9: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  10: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  11: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  12: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  13: P.2: exits normally
  14: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  15: P.3: exits normally
  16: Message (ok2) from P.2 reaches P
  17: P: receives message (ok2)
    in after_test.erl line 26
  18: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  19: Message (ok2) from P.3 reaches P
  20: P: receives message (ok2)
    in after_test.erl line 22
  21: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 19:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   5: P.1: exits normally
   6: Message (ok1) from P.1 reaches P
   7: P: receives message (ok1)
    in after_test.erl line 26
   8: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   9: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  10: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  11: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  12: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  13: P.2: exits normally
  14: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  15: P.3: exits normally
  16: Message (ok2) from P.2 reaches P
  17: P: receives message (ok2)
    in after_test.erl line 26
  18: Message (ok2) from P.3 reaches P
  19: P: receives message (ok2)
    in after_test.erl line 26
  20: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 20:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   5: P.1: exits normally
   6: Message (ok1) from P.1 reaches P
   7: P: receives message (ok1)
    in after_test.erl line 26
   8: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   9: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  10: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  11: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  12: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  13: P.2: exits normally
  14: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  15: P.3: exits normally
  16: Message (ok2) from P.3 reaches P
  17: P: receives message (ok2)
    in after_test.erl line 26
  18: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  19: Message (ok2) from P.2 reaches P
  20: P: receives message (ok2)
    in after_test.erl line 22
  21: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 21:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   5: P.1: exits normally
   6: Message (ok1) from P.1 reaches P
   7: P: receives message (ok1)
    in after_test.erl line 26
   8: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   9: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  10: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  11: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  12: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  13: P.2: exits normally
  14: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  15: P.3: exits normally
  16: Message (ok2) from P.3 reaches P
  17: P: receives message (ok2)
    in after_test.erl line 26
  18: Message (ok2) from P.2 reaches P
  19: P: receives message (ok2)
    in after_test.erl line 26
  20: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 22:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   5: P.1: exits normally
   6: Message (ok1) from P.1 reaches P
   7: P: receives message (ok1)
    in after_test.erl line 26
   8: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   9: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  10: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  11: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  12: P.2: exits normally
  13: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  14: P.3: exits normally
  15: Message (ok2) from P.2 reaches P
  16: Message (ok2) from P.3 reaches P
  17: P: receives message (ok2)
    in after_test.erl line 26
  18: P: receives message (ok2)
    in after_test.erl line 26
  19: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 23:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   5: P.1: exits normally
   6: Message (ok1) from P.1 reaches P
   7: P: receives message (ok1)
    in after_test.erl line 26
   8: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   9: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  10: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  11: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  12: P.2: exits normally
  13: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  14: P.3: exits normally
  15: Message (ok2) from P.2 reaches P
  16: P: receives message (ok2)
    in after_test.erl line 26
  17: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  18: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  19: Message (ok2) from P.3 reaches P
  20: P: receives message (ok2)
    in after_test.erl line 22
  21: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 24:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   5: P.1: exits normally
   6: Message (ok1) from P.1 reaches P
   7: P: receives message (ok1)
    in after_test.erl line 26
   8: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   9: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  10: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  11: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  12: P.2: exits normally
  13: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  14: P.3: exits normally
  15: Message (ok2) from P.2 reaches P
  16: P: receives message (ok2)
    in after_test.erl line 26
  17: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  18: Message (ok2) from P.3 reaches P
  19: P: receives message (ok2)
    in after_test.erl line 26
  20: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 25:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   5: P.1: exits normally
   6: Message (ok1) from P.1 reaches P
   7: P: receives message (ok1)
    in after_test.erl line 26
   8: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   9: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  10: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  11: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  12: P.2: exits normally
  13: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  14: P.3: exits normally
  15: Message (ok2) from P.2 reaches P
  16: P: receives message (ok2)
    in after_test.erl line 26
  17: Message (ok2) from P.3 reaches P
  18: P: receives message (ok2)
    in after_test.erl line 26
  19: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 26:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   5: P.1: exits normally
   6: Message (ok1) from P.1 reaches P
   7: P: receives message (ok1)
    in after_test.erl line 26
   8: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   9: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  10: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  11: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  12: P.2: exits normally
  13: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  14: P.3: exits normally
  15: Message (ok2) from P.3 reaches P
  16: P: receives message (ok2)
    in after_test.erl line 26
  17: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  18: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  19: Message (ok2) from P.2 reaches P
  20: P: receives message (ok2)
    in after_test.erl line 22
  21: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 27:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   5: P.1: exits normally
   6: Message (ok1) from P.1 reaches P
   7: P: receives message (ok1)
    in after_test.erl line 26
   8: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   9: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  10: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  11: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  12: P.2: exits normally
  13: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  14: P.3: exits normally
  15: Message (ok2) from P.3 reaches P
  16: P: receives message (ok2)
    in after_test.erl line 26
  17: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  18: Message (ok2) from P.2 reaches P
  19: P: receives message (ok2)
    in after_test.erl line 26
  20: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 28:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P: receive timeout expired after 0 ms
    in after_test.erl line 26
   4: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   5: P.1: exits normally
   6: Message (ok1) from P.1 reaches P
   7: P: receives message (ok1)
    in after_test.erl line 26
   8: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   9: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
  10: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  11: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  12: P.2: exits normally
  13: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  14: P.3: exits normally
  15: Message (ok2) from P.3 reaches P
  16: P: receives message (ok2)
    in after_test.erl line 26
  17: Message (ok2) from P.2 reaches P
  18: P: receives message (ok2)
    in after_test.erl line 26
  19: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 29:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   4: P.1: exits normally
   5: Message (ok1) from P.1 reaches P
   6: P: receives message (ok1)
    in after_test.erl line 26
   7: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   8: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
   9: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  10: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  11: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  12: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  13: P.2: exits normally
  14: Message (ok2) from P.2 reaches P
  15: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  16: P.3: exits normally
  17: Message (ok2) from P.3 reaches P
  18: P: receives message (ok2)
    in after_test.erl line 22
  19: P: receives message (ok2)
    in after_test.erl line 22
  20: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 30:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   4: P.1: exits normally
   5: Message (ok1) from P.1 reaches P
   6: P: receives message (ok1)
    in after_test.erl line 26
   7: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   8: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
   9: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  10: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  11: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  12: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  13: P.2: exits normally
  14: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  15: P.3: exits normally
  16: Message (ok2) from P.3 reaches P
  17: P: receives message (ok2)
    in after_test.erl line 22
  18: Message (ok2) from P.2 reaches P
  19: P: receives message (ok2)
    in after_test.erl line 22
  20: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 31:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   4: P.1: exits normally
   5: Message (ok1) from P.1 reaches P
   6: P: receives message (ok1)
    in after_test.erl line 26
   7: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   8: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
   9: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  10: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  11: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  12: P.2: exits normally
  13: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  14: P.3: exits normally
  15: Message (ok2) from P.2 reaches P
  16: Message (ok2) from P.3 reaches P
  17: P: receives message (ok2)
    in after_test.erl line 26
  18: P: receives message (ok2)
    in after_test.erl line 26
  19: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 32:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   4: P.1: exits normally
   5: Message (ok1) from P.1 reaches P
   6: P: receives message (ok1)
    in after_test.erl line 26
   7: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   8: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
   9: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  10: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  11: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  12: P.2: exits normally
  13: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  14: P.3: exits normally
  15: Message (ok2) from P.2 reaches P
  16: P: receives message (ok2)
    in after_test.erl line 26
  17: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  18: Message (ok2) from P.3 reaches P
  19: P: receives message (ok2)
    in after_test.erl line 22
  20: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 33:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   4: P.1: exits normally
   5: Message (ok1) from P.1 reaches P
   6: P: receives message (ok1)
    in after_test.erl line 26
   7: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   8: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
   9: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  10: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  11: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  12: P.2: exits normally
  13: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  14: P.3: exits normally
  15: Message (ok2) from P.2 reaches P
  16: P: receives message (ok2)
    in after_test.erl line 26
  17: Message (ok2) from P.3 reaches P
  18: P: receives message (ok2)
    in after_test.erl line 26
  19: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 34:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   4: P.1: exits normally
   5: Message (ok1) from P.1 reaches P
   6: P: receives message (ok1)
    in after_test.erl line 26
   7: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   8: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
   9: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  10: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  11: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  12: P.2: exits normally
  13: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  14: P.3: exits normally
  15: Message (ok2) from P.3 reaches P
  16: P: receives message (ok2)
    in after_test.erl line 26
  17: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  18: Message (ok2) from P.2 reaches P
  19: P: receives message (ok2)
    in after_test.erl line 22
  20: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 35:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   4: P.1: exits normally
   5: Message (ok1) from P.1 reaches P
   6: P: receives message (ok1)
    in after_test.erl line 26
   7: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   8: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
   9: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  10: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  11: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  12: P.2: exits normally
  13: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  14: P.3: exits normally
  15: Message (ok2) from P.3 reaches P
  16: P: receives message (ok2)
    in after_test.erl line 26
  17: Message (ok2) from P.2 reaches P
  18: P: receives message (ok2)
    in after_test.erl line 26
  19: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 36:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   4: P.1: exits normally
   5: Message (ok1) from P.1 reaches P
   6: P: receives message (ok1)
    in after_test.erl line 26
   7: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   8: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
   9: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  10: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  11: P.2: exits normally
  12: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  13: P.3: exits normally
  14: Message (ok2) from P.2 reaches P
  15: Message (ok2) from P.3 reaches P
  16: P: receives message (ok2)
    in after_test.erl line 26
  17: P: receives message (ok2)
    in after_test.erl line 26
  18: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 37:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   4: P.1: exits normally
   5: Message (ok1) from P.1 reaches P
   6: P: receives message (ok1)
    in after_test.erl line 26
   7: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   8: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
   9: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  10: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  11: P.2: exits normally
  12: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  13: P.3: exits normally
  14: Message (ok2) from P.2 reaches P
  15: P: receives message (ok2)
    in after_test.erl line 26
  16: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  17: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  18: Message (ok2) from P.3 reaches P
  19: P: receives message (ok2)
    in after_test.erl line 22
  20: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 38:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   4: P.1: exits normally
   5: Message (ok1) from P.1 reaches P
   6: P: receives message (ok1)
    in after_test.erl line 26
   7: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   8: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
   9: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  10: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  11: P.2: exits normally
  12: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  13: P.3: exits normally
  14: Message (ok2) from P.2 reaches P
  15: P: receives message (ok2)
    in after_test.erl line 26
  16: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  17: Message (ok2) from P.3 reaches P
  18: P: receives message (ok2)
    in after_test.erl line 26
  19: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 39:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   4: P.1: exits normally
   5: Message (ok1) from P.1 reaches P
   6: P: receives message (ok1)
    in after_test.erl line 26
   7: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   8: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
   9: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  10: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  11: P.2: exits normally
  12: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  13: P.3: exits normally
  14: Message (ok2) from P.2 reaches P
  15: P: receives message (ok2)
    in after_test.erl line 26
  16: Message (ok2) from P.3 reaches P
  17: P: receives message (ok2)
    in after_test.erl line 26
  18: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 40:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   4: P.1: exits normally
   5: Message (ok1) from P.1 reaches P
   6: P: receives message (ok1)
    in after_test.erl line 26
   7: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   8: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
   9: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  10: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  11: P.2: exits normally
  12: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  13: P.3: exits normally
  14: Message (ok2) from P.3 reaches P
  15: P: receives message (ok2)
    in after_test.erl line 26
  16: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  17: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  18: Message (ok2) from P.2 reaches P
  19: P: receives message (ok2)
    in after_test.erl line 22
  20: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 41:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   4: P.1: exits normally
   5: Message (ok1) from P.1 reaches P
   6: P: receives message (ok1)
    in after_test.erl line 26
   7: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   8: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
   9: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  10: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  11: P.2: exits normally
  12: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  13: P.3: exits normally
  14: Message (ok2) from P.3 reaches P
  15: P: receives message (ok2)
    in after_test.erl line 26
  16: P: receive timeout expired after 0 ms
    in after_test.erl line 26
  17: Message (ok2) from P.2 reaches P
  18: P: receives message (ok2)
    in after_test.erl line 26
  19: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Erroneous interleaving 42:
* Blocked at a 'receive' (when all other processes have exited):
    P in after_test.erl line 16
--------------------------------------------------------------------------------
Interleaving info:
   1: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   2: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test.0.130445682>,[]])
    in erlang.erl line 2316
   3: P.1: ok1 = P ! ok1
    in after_test.erl line 11
   4: P.1: exits normally
   5: Message (ok1) from P.1 reaches P
   6: P: receives message (ok1)
    in after_test.erl line 26
   7: P: receive timeout expired after 0 ms
    in after_test.erl line 33
   8: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test.1.130445682>,[]])
    in erlang.erl line 2316
   9: P: P.3 = erlang:spawn(erlang, apply, [#Fun<after_test.2.130445682>,[]])
    in erlang.erl line 2316
  10: P.2: ok2 = P ! ok2
    in after_test.erl line 13
  11: P.2: exits normally
  12: P.3: ok2 = P ! ok2
    in after_test.erl line 14
  13: P.3: exits normally
  14: Message (ok2) from P.3 reaches P
  15: P: receives message (ok2)
    in after_test.erl line 26
  16: Message (ok2) from P.2 reaches P
  17: P: receives message (ok2)
    in after_test.erl line 26
  18: P: receive timeout expired after 0 ms
    in after_test.erl line 33
################################################################################
Tips:
--------------------------------------------------------------------------------
Every interleaving explored so far had some error. This can make later debugging difficult, as the generated report will include too much info. You may want to consider refactoring your code, or using the appropriate options to filter out irrelevant errors.
################################################################################
Info:
--------------------------------------------------------------------------------
Instrumented after_test
Instrumented io_lib
Instrumented erlang
################################################################################
Done! (Exit status: completed)
  Summary: 42 errors, 42/42 interleavings explored
